tests/unittests/GraphOptzTest.cpp

```diff
@@ -495,35 +495,75 @@ TEST_F(GraphOptz, batchNormAfterConvNotOptimizeWhenMoreThanOneUseOfConv) {
   EXPECT_EQ(conv->getInput().getNode(), A);
 }
 
-TEST_F(GraphOptz, sinkTransposeBelowOptimizeBatchNorm) {
+enum class TestSinkTransposeNodesKind {
+  Batch,
+  Relu,
+  Sigmoid,
+  Tanh,
+  Quantize,
+};
+
+class GraphOptzSinkTransposeBelowParametrized
+    : public GraphOptz,
+      public ::testing::WithParamInterface<TestSinkTransposeNodesKind> {
+public:
+  Node *getNodeFromInput(TestSinkTransposeNodesKind testNode, Node *T) {
+    switch (testNode) {
+    case TestSinkTransposeNodesKind::Batch: {
+      return F_->createBatchNormalization(bindings_, "batch", T, 3, 0.0001,
+                                          0.9);
+    }
+    case TestSinkTransposeNodesKind::Relu: {
+      return F_->createRELU("relu", T);
+    }
+    case TestSinkTransposeNodesKind::Sigmoid: {
+      return F_->createSigmoid("sigmoid", T);
+    }
+    case TestSinkTransposeNodesKind::Tanh: {
+      return F_->createTanh("tanh", T);
+    }
+    case TestSinkTransposeNodesKind::Quantize: {
+      return F_->createQuantize(
+          "quantize", T,
+          mod_.uniqueType(ElemKind::Int8QTy, T->dims(0), 0.03, 5));
+    }
+    default: { return nullptr; }
+    }
+  };
+};
+
+TEST_P(GraphOptzSinkTransposeBelowParametrized,
+       TestSinkTransposeForDifferentCases) {
   const size_t origDims[] = {1, 5, 10, 15};
   const size_t transposedDims[] = {1, 15, 5, 10};
   Node *A = mod_.createPlaceholder(ElemKind::FloatTy, origDims, "input", false);
   Node *T = F_->createTranspose("transpose", A, NHWC2NCHW);
-  BatchNormalizationNode *BN =
-      F_->createBatchNormalization(bindings_, "batch", T, 3, 0.0001, 0.9);
-  SaveNode *O = F_->createSave("ret", BN);
+  Node *N = getNodeFromInput(GetParam(), T);
+  SaveNode *O = F_->createSave("ret", N);
 
   EXPECT_EQ(F_->getNodes().size(), 3);
-  EXPECT_EQ(BN->getResult().dims(), llvm::makeArrayRef(transposedDims));
+  EXPECT_EQ(N->getNthResult(0).dims(), llvm::makeArrayRef(transposedDims));
 
   ::glow::optimize(F_, CompilationMode::Infer);
 
-  // Expecting Transpose->Output rather than BN->Output.
+  // Expecting Transpose->Output rather than N->Output.
   auto *transpose = llvm::dyn_cast<TransposeNode>(O->getInput());
   ASSERT_NE(transpose, nullptr);
-  auto *bn = llvm::dyn_cast<BatchNormalizationNode>(transpose->getInput());
-  ASSERT_TRUE(bn);
+  auto *n = llvm::dyn_cast<Node>(transpose->getInput());
+  ASSERT_TRUE(n);
   // Check that the dimensions of the input and output have been
   // updated to compensate the absence of transpose.
-  EXPECT_EQ(bn->getResult().dims(), llvm::makeArrayRef(origDims));
-  EXPECT_EQ(bn->getInput().dims(), llvm::makeArrayRef(origDims));
+  EXPECT_EQ(n->getNthResult(0).dims(), llvm::makeArrayRef(origDims));
+  EXPECT_EQ(n->getNthInput(0).dims(), llvm::makeArrayRef(origDims));
```

Comment:
This one is a bit tougher 🙁I'd at least prefer to add a check here to make sure we're testing the correct input, especially since we could add new cases to this test. E.g.:
```
assert(n->getNumInputs() == 1 || 
       (GetParam() == TestSinkTransposeNodesKind::Batch && 
        BatchNormalization::InputIdx == 0));
```